docs/users-guide/custom-questions.md

```diff
@@ -0,0 +1,126 @@
+
+Now we can use this new column just like any other one, meaning we can use it to filter our question, add a grouping with it, or find out things like the average of it. You can add multiple custom fields, and they’ll all show up at the top of drop downs within the question builder:
+
+![Field in dropdown](images/custom-fields/field-in-dropdown.png)
+
+#### Sorting results
+
+![Sorting](./images/notebook/sort-step.png)
+
+The sorting step lets you pick one or more columns to sort your results by. For each column you pick, you can also choose whether to sort ascending or descending; just click the arrow to change from ascending (up arrow) to descending (down arrow).
+
+#### Setting a row limit
+
+The row limit step lets you limit how many rows you want from the previous results. When used in conjunction with sorting, this can let you do things like create a top-10 list, by first sorting by one of the columns in your result, then adding a row limit of 10. Unlike other steps, the row limit step can only be added at the end of your question.
+
+#### Joining data
+
+![Joining](./images/notebook/join-step.png)
+
+The join step allows you to combine your current data with another table, or even with a saved question.
+
+Currently you can't use joins if your starting data is from a Google Analytics or MongoDB database.
+
+After you click on the Join Data button to add a join step, you'll need to pick the data that you want to join. **Note:** You can only pick tables and saved questions that are from the same database as your starting data.
+
+![Picking the data to join](./images/notebook/join-pick-data.png)
+
+Next, you'll need to pick the columns you want to join on. This means you pick a column from the first table, and a column from the second table, and the join will stitch rows together where the value from the first column is equal to the value in the second column. A very common example is to join on an ID column in each table, so if you happened to pick a table to join on where there is a foreign key relationship between the tables, Metabase will automatically pick those corresponding ID columns for you. At the end of your join step, there's a `Columns` button you can click to choose which columns you want to include from the joined data.
+
+By default, Metabase will do a left outer join, but you can click on the Venn diagram icon to change this to a different type of join. The options you'll see will differ based on the type of database you're using. Here are what the basic types of joins each do:
+
+- **Left outer join:** select all records from Table A, along with records from Table B that meet the join condition, if any.
+- **Right outer join:** select all records from Table B, along with records from Table B that meet the join condition, if any.
+- **Inner join:** only select the records from Table A and B where the join condition is met.
+- **Full outer join:** select all records from both tables, whether or not the join condition is met.
```

Comment:
"Full outer join" - when is this option available?

Reply:
Not available for MySQL, but it should be available for Spark SQL, SQLite, and SQL Server I think.